tests/cmake_test.cc

```diff
@@ -0,0 +1,18 @@
+#include <glib.h>
+#include "cmake.h"
+#include "project_build.h"
+#include <boost/filesystem.hpp>
+
+int main() {
+  CMake cmake(JUCI_TESTS_PATH);
+  
+  g_assert(cmake.project_path.filename()=="jucipp");
+  
```

Comment:
Well, don't rename it;)? It's just a sample test though, can probably be improved. This PR is mostly a proof of concept, and directions and samples for creating future tests.